README.md: Include a bit of real documentation (#31)
Jovian-Experiments/Jovian-NixOS: README.md

NixOS on the Steam Deck

> =======================

Reviewer: ```suggestion
<div align="center"><h1>Jovian NixOS</h1></div>
<div align="center"><strong><em>NixOS on the Steam Deck</em></strong></div>

****

```

How does that sound?

 - Let's center the title.
 - Let's make the 5 word pitch a subtitle.

NixOS on the Steam Deck
=======================

> **Jovian**  

> > “Relating to [...] Jupiter or the class of [...] which Jupiter belongs.”

Reviewer: I like to finish READMEs with FAQs, and here since the name *can* be odd if you don't know the definition for *Jovian*, it would be nice to add this snippet back in.

> This repo is also useful for obtaining a Deck-like experience on other X86-64 devices.

Reviewer: ```suggestion
This repo is also useful for obtaining a Deck-like experience on other x86_64 devices.
```

Format used by e.g.  `uname` and by convention system types in Nix and Nixpkgs.

Or even `x86_64` (with backticks to make it `<code />`)

------------

You can build an installation ISO for the Steam Deck with `nix-build -A <flavor>`, where `<flavor>` is one of the following:

- `isoMinimal`: Minimal non-graphical installation environment
- `isoGnome`: Graphical installation ISO with GNOME
- `isoPlasma`: Graphical installation ISO with KDE

To enter the boot menu, power off the Steam Deck then hold down `Volume-` and tap the `Power` button.

> Follow the [Installing NixOS](https://nixos.org/manual/nixos/unstable/index.html#sec-installation) section of the NixOS Manual.
> When configuring the system, import `./modules` from this repo in your configuration, then enable the Steam Deck-specific hardware configurations with `jovian.devices.steamdeck.enable = true;`.

Reviewer: ```suggestion
Follow the [Installing NixOS](https://nixos.org/manual/nixos/unstable/index.html#sec-installation) section of the NixOS Manual.
When configuring the system, import `./modules` from this repo in your configuration.

When used on a Steam Deck, enable the Steam Deck-specific hardware configurations with `jovian.devices.steamdeck.enable = true;`.
```

Small nit since we're stating it can also be used on other hardware than a Steam Deck. Let's make it really more obvious that this is only for Steam Deck hardware.

Author: Since our ISO shouldn't be used on other hardware, let's try splitting the section to make it less hairy.
